src/components/Tasks/Sidebar.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+
+import { useSelector, useDispatch } from 'react-redux';
+
+import List from '@material-ui/core/List';
+import ListSubheader from '@material-ui/core/ListSubheader';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import { IconProps } from '@material-ui/core/Icon';
+import LabelIcon from '@material-ui/icons/LabelOutlined';
+
+import { setSettings } from '../../store/actions';
+import { StoreState } from '../../store';
+
+interface ListItemPropsType {
+  name: string | null;
+  Icon?: React.ComponentType<IconProps>;
+  primaryText: string;
+  secondaryText?: string;
+}
+
+function SidebarListItem(props: ListItemPropsType) {
+  const { name, Icon, primaryText, secondaryText } = props;
+  const dispatch = useDispatch();
+  const taskSettings = useSelector((state: StoreState) => state.settings.taskSettings);
+  const { filterBy } = taskSettings;
+
+  const handleClick = () => {
+    dispatch(setSettings({ taskSettings: { ...taskSettings, filterBy: name } }));
+  };
+
+  return (
+    <ListItem
+      button
+      onClick={handleClick}
+      selected={name === filterBy}
+    >
+      {Icon ?
+        <ListItemIcon>
+          <Icon fontSize="small" />
+        </ListItemIcon> :
+        ''
+      }
+      <ListItemText primary={primaryText} secondary={secondaryText || ''} />
+    </ListItem>
+  );
+}
+
+export default function Sidebar(props: { tags: Map<string, number> }) {
+  const { tags } = props;
+
+  const tagsList = Array.from(tags, ([tag, amount]) => (
+    <SidebarListItem
+      key={tag}
+      name={`tag ${tag}`}
+      primaryText={tag}
+      secondaryText={String(amount)}
```

Comment:
I think it's better to just pass the numerical amount to the `SidebarListItem` and then let it decide how to design itself. Here you are kind of forcing it from the outside to put it in the secondaryText field which feels off.

The reason why I even noticed this forcing from the outside, is because I think the count looks quite odd alone there on its own line. Maybe have it as a pill (badge?) next to the name? So it would look something like: `Work [2]` where the square brackets represent a pill/badge?
Or maybe follow the example in #32 and just have it aligned to the right (and make the sidebar larger)?

Reply:
I've changed the design to match #32 more closely. I had to do some finagling with passing styles down to the `ListItem` component. I'm not sure if I did this correctly, or if this is the best way to do it, so please let me know your thoughts.

Reply:
I think it's a bit hacky, and commented there.